core/trino-main/src/test/java/io/trino/operator/TestTopNOperator.java

```diff
@@ -198,13 +198,10 @@ public void testExceedMemoryLimit()
                 100,
                 ImmutableList.of(0),
                 ImmutableList.of(ASC_NULLS_LAST));
-        try (Operator operator = operatorFactory.createOperator(smallDiverContext)) {
-            operator.addInput(input.get(0));
-            operator.getOutput();
-            fail("must fail because of exceeding local memory limit");
-        }
-        catch (ExceededMemoryLimitException ignore) {
-        }
+        Operator operator = operatorFactory.createOperator(smallDiverContext);
+        assertThatThrownBy(() -> operator.addInput(input.get(0)))
+                .isInstanceOf(ExceededMemoryLimitException.class)
+                .hasMessageStartingWith("Query exceeded per-node user memory limit of ");
     }
```

Comment:
in the previous version operator was created inside try{}, which is more safe, may be it is better to keep this?